buildSrc/src/main/groovy/MuzzlePlugin.groovy

```diff
@@ -95,10 +95,7 @@ class MuzzlePlugin implements Plugin<Project> {
     project.tasks.compileMuzzle.dependsOn(bootstrapProject.tasks.compileJava)
     project.tasks.compileMuzzle.dependsOn(toolingProject.tasks.compileJava)
     project.afterEvaluate {
-      project.tasks.compileMuzzle.dependsOn(project.tasks.compileJava)
-      if (project.tasks.getNames().contains('compileScala')) {
-        project.tasks.compileMuzzle.dependsOn(project.tasks.compileScala)
-      }
+      project.tasks.compileMuzzle.dependsOn(project.tasks.classes)
```

Comment:
This seems to be the actual fix of our muzzle problems - what does it change? We still use `compileJava` two lines above, is that OK?

Reply:
Yes, toolingProject and bootstrapProject are Ok to use `compileJava`, although we may want to change that to `classes` as well for consistency. The problem is that in instrumentation modules we use a separate bytebuddy compilation plugin. And that plugin messes with `compileJava` task. Using `classes` instead guarantees that byte buddy ran as well.

Reply:
Can you add a comment about that? That was kind of non-obvious fix

Reply:
I simplified and unified this.